aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -224,6 +224,8 @@ def _generate_hash(self, data: Any) -> str:
             Hashed representation of the provided data
 
         """
+        if hasattr(data, "raw_event"):
+            data = data.raw_event
```

Comment:
`hasattr` doesn't work in the way you might think it does - it's a misnomer in Python, really. It'll effectively do a blanket try/except, throw the results away, and specially in this case it'll access all getters from `data` -- the more getters the worse the performance.

Here's a safer alternative which also saves us a branching:

```suggestion
data = getattr(data, "raw_event", data)  # could be a data class depending on decorator order
```